src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+        return topic;
+    }
+
+    function shouldCollectReportingFees() public constant returns (bool) {
+        return !getBranch().isContainerForShareToken(cash);
+    }
+
+    function isDoneWithAutomatedReporters() public constant returns (bool) {
+        return automatedReportReceived || block.timestamp > getAutomatedReportDueTimestamp();
+    }
+
+    function isDoneWithLimitedReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.LIMITED_REPORTING;
+    }
+
+    function isDoneWithAllReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.ALL_REPORTING;
+    }
+
+    function isFinalized() public constant returns (bool) {
+        return finalPayoutDistributionHash != bytes32(0);
+    }
+
+    function getFinalizationTime() public constant returns (uint256) {
+        return finalizationTime;
+    }
+
+    function isInAutomatedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_REPORTING;
+    }
+
+    function isInAutomatedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_DISPUTE;
+    }
+
+    function isInLimitedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_REPORTING;
+    }
+
+    function isInLimitedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_DISPUTE;
+    }
+
+    function isInAllReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_REPORTING;
+    }
+
+    function isInAllDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_DISPUTE;
+    }
+
+    function isContainerForReportingToken(ReportingToken _shadyToken) public constant returns (bool) {
+        if (address(_shadyToken) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_shadyToken.getTypeName() != "ReportingToken") {
+            return false;
+        }
+        bytes32 _shadyId = _shadyToken.getPayoutDistributionHash();
+        ReportingToken _reportingToken = reportingTokens[_shadyId];
+        if (address(_reportingToken) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_reportingToken != _shadyToken) {
+            return false;
+        }
+        return true;
+    }
+
+    function isContainerForShareToken(IShareToken _shadyShareToken) public constant returns (bool) {
+        if (_shadyShareToken.getTypeName() != "ShareToken") {
+            return false;
+        }
+        return(getShareToken(_shadyShareToken.getOutcome()) == _shadyShareToken);
+    }
+
+    function isContainerForDisputeBondToken(DisputeBondToken _shadyBondToken) public constant returns (bool) {
+        if (_shadyBondToken.getTypeName() != "DisputeBondToken") {
+            return false;
+        }
```

Comment:
The type check here is unnecessary since we won't be calling any methods on the contract.